src/main/java/org/eclipse/opensmartclide/dbapi/model/Deployment.java

```diff
@@ -63,10 +67,9 @@ public class Deployment {
     //@DocumentReference
     //private Service service;
 
-    @NotNull
     private String version;
 
-    //@NotNull
+    @NotNull
```

Comment:
I think that the NotNull annotation is not enough here.

Actually, NotNull means that a field cannot be null, but it can be empty. We should ensure that the field should not be empty as well.

So, I think it is better to use the NotEmpty annotation.

From [here](https://docs.jboss.org/hibernate/beanvalidation/spec/2.0/api/):

Annotation Type NotEmpty: The annotated element must not be null nor empty.